### PR TITLE
A new blog post on Impostor Syndrome

### PR DESCRIPTION
**Pull request by crystoll**
A short article on software consulting and impostor syndrome. Any feedback is welcome!

### CODE REVIEW

**Comment by crystoll** [comment]
And actually at least 1 approved review is required :)

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title: Generalist Specialization and The Impostor Syndrome
+author: arto
+excerpt: >
+  Impostor Syndrome is a common visitor in our profession, as we are often in charge of shaping the world of the future, while boldly going to new, unknown areas. In this article I'll touch on what it is, how to recognize it, what to do about it, and how does it relate to being a generalist.
+tags:
+ - Software development
```
Comment (tuk-solita) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:8]: The tag is case-sensitive. Right now there is five posts tagged as "Software Development" and one as "Software development". Just something to consider.

**Review by tuk-solita — APPROVED** [review]
Looks and reads good to me!

**Review by tuk-solita — APPROVED** [review]
What the heck. How do I "review"

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md (hunk 1 above)
Comment (crystoll) [reply _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:8]: Thanks, that was a good catch. I'll fix that.

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Neglecting any of these, in a professionally done project, can doom it to failure, or at least make it very difficult. So to be able to do your work properly, there's an endless stream of things you need to be aware of and make the right choices. The more you know the more you know what you don't know.
+
+Furthermore, things move rapidly. While you can use your knowledge from decades back, in our line of work we typically need to be constantly learning more, keeping skills up to date, replacing old tools with new, better ones. While at some point in time you could say you know everything there is about programming, now things have exploded exponentially. If anyone tells you they know all there is to know about software delivery, they are lying.
+
+So, that's the scene. That's the setup. Things might be a bit calmer if you're working on the same product/service for years, but I can tell you that especially in consulting business, you have to push yourself to new areas constantly, and quickly master them. Many people actually enjoy that, on a good day it's a rush to be able to grow and learn new things.
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:26]: "if you work on"

"in *the* consulting business"

"… enjoy that. On a good…"

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+So, while specialization is still rather a valid thing to do, you should also invest in learning a wide array of skills for software crafting, as listed above. These days you definitely need to understand automated testing, delivery, security, and cloud platforms, in addition to Frontend, Backend, and databases. But you can still specialize, you can double on on one or two things that bring you pleasure. You can actively practice them, learn more about them, and be the go-to-person for said techniques.
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:36]: Listed where? Do you mean the list of programming languages and AWS?

No need to capitalize frontend and backend.

"double down on"

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+But there's also another route. I'm a Java coder. That's been my forte. I started it in the 1990s, then I did a long period of teaching it for money, which meant I needed to first master it myself, and keep up to date. I would say Java is something I can code in my sleep, and sometimes do, so that's definitely my specialization and strength. But would you believe it if I said I have barely used any Java for several years now? I've been relying on my 25 years of experience on all things, and my ability to rapidly pick up new things based on that. If I drew my skills profile right now it would be crazy. I can do many things. Some might argue even any things. My recent years have been filled for example by Splunk and Excel coding, a lot of Python, Scala, some Kotlin, Gitlab pipelines, and lots and lots of cloud acronyms. Did you know that AWS has hundreds of services and thousands of acronyms? And each one is a specialization in itself.
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:38]: "Gitlab" -> "GitLab"

**Review by tatut — APPROVED** [review]

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+So I would call myself a generalist. That's been my specialization. Meaning that I actively pursue that and keep my generalization up to date. I know few things really in-depth, but I know a hundred things on a level that I can rapidly refresh up to today's requirements if there's a need. A few years back someone asked me about XSLT templates, so I whipped up from my ancient technologies archive a nice recursion template that lets you do crazy stuff with minimal code. That gives me much flexibility to jump in many places. Unfortunately, that comes with a downside too. A constant visitor.
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:40]: "whipped up" means "create", "whipped out" is probably more  appropriate here. Also, yay for XSLT!

"That gives" -- "That" refers to the XSLT snippet. Do you mean to refer to the fact that you're a generalist? Perhaps "My generalism gives..."

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+## How can you deal with the Impostor Syndrome?
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:42]: "with impostor syndrome?"

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+There are two bad ways to try to deal with this. One is to never put yourself in a position of change or challenge. Just play it very safe. You learned Java 1.0 in 1996, so keep on using it for everything, and only look for jobs where that is all that is required. This approach would let the Impostor Syndrome rule and limit you. Another is to wait, it will pass. But that's not going to help you deal with it any better next time, so while it may be a victory, it's a lame one.
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:44]: "with this" -> "with it" or "with impostor syndrome".

"…let impostor syndrome rule…"

"another strategy is to wait"

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+A better approach is just to acknowledge that it is a thing, and recognize it. Understand that if you feel like that, it does not mean you actually are an impostor. But most of us have studied hard, have practiced hard, have worked hard, more or less. So the background is there, the skills are there. That feeling is not a real reflection of things.
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:46]: "Understand that feeling like an impostor doesn't make you one" maybe?

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+It's also not a mental disorder, nor a permanent state of mind. It's a reaction to stimuli, especially when you are high-performing and reaching far. It could be compared to driving a sports car at high speeds, and suddenly having a doubt if you can control it if you actually have the skills for that. It's a temporary moment of self-doubt, that often happens at the moment when you have achieved (or are about to achieve) an important milestone. New job, new project, a new position, or just praise and adoration for the work you have done. Ironically, this can also be triggered by positive feedback and compliments.
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:48]: "suddenly having a doubt if you can control it if you actually have the skills for that." -- is there a word missing here?

Perhaps something like "suddenly starting to doubt whether you can actually control it" .

"self-doubt, that" -> "self-doubt that" (there's almost never a comma before "that")

"happens ~at the moment~ when" -- "at the moment" + "when" == redundant

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+Also, note that typically this moment of self-doubt also means that at your worst moment you're trying to compare yourself to other people at their best moment. Or even worse, trying to compare yourself to an ideal image of yourself. It's an unfair advantage, and one you will always lose. There is always someone who knows more about something.
+
+According to some studies made on the subject, the main medication is just to try to reframe/rephrase that self-doubt with a bit more bright way of looking at things. For example, if your thought is: "I'm out of my league, I don't know how to do this!" - then you could try to reframe it like "This seems difficult, but I have time to educate myself, and everybody understands I'm not a master of everything. I am allowed to fail."
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:52]: "a bit more bright way" - "a brighter way"

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+According to some studies made on the subject, the main medication is just to try to reframe/rephrase that self-doubt with a bit more bright way of looking at things. For example, if your thought is: "I'm out of my league, I don't know how to do this!" - then you could try to reframe it like "This seems difficult, but I have time to educate myself, and everybody understands I'm not a master of everything. I am allowed to fail."  
+
+This actually goes nicely with agile methodologies, too, because a failure IS an option - you just try to fail fast in that case, take your learnings, and build on top of that. Transparency also helps - the feeling of being a fraud often comes if you hide your internal worries deep, and try to set up a facade. As with any self-doubt, it might do marvels if you just have a chat with your peers, or project lead, or life coach, and are honest about your worries.
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:54]: "This" -> maybe "This line of thinking"?

"in that case" is unnecessary

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+So if you feel unsure, stop for a bit, take a deep breath, and try to see things from another angle. Try this question:
+
+**"What would I need to change to be able to succeed?"**
+
+## Impostor syndrome as a positive thing?
+
+The analytical part of my brain understands all this clearly. But unfortunately, feelings are not something that can be dismissed by logical thought and willpower alone. One way of course to avoid those feelings would be to stop pushing yourself into those situations anymore. If you know one thing well, stick to that one thing? Well, while you certainly could do that, I would not recommend it.
+
+See, growth comes from the combination of applying what you know but being also exposed to things you don't know - yet. At least for me, that might raise the appetite to know more, to understand more, to learn more. And that is half of why I love the software developer life (The other half is instant gratification from solving concrete problems). As I recognize that feeling for what it is, I can avoid letting it overwhelm me or bring stress on me.
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:64]: "being also" -> "also being"

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+So for me, the impostor syndrome visiting is actually a signal that I've once again managed to push myself to a new uncomfortable zone, where I don't already know everything. So I've managed to create a learning and growth opportunity for this year, too. And that's exactly why I mean it when I say:
```
Comment (eerohele) [inline _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:66]: "the impostor syndrome visiting" -> "a visit from impostor syndrome"

"zone, where" -> "zone where"

**Review by eerohele — COMMENTED** [review]
Great post! 👍🏻

File: _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+![Feeling of self-doubt](/img/impostor-syndrome/doubt.jpg)
+
+_I don't know how to do this. This is all too overwhelming. Why did I want to do this? Everybody else knows what they are doing. They will soon discover how out of my league I am here._
+
+Sounds familiar? Maybe not. Not everyone is suffering from Impostor Syndrome. But it is quite common in our field of work. I've had those thoughts myself, and many times have also seen others go through moments like this. It's not typically a permanent state in life, but join me for a bit and let's talk about it.
+
+## Why do we feel like impostors?
+
+Creating quality software in 2021 is hard. There are so many facets to consider in addition to just writing the code. Things like the choice of programming languages, frameworks, and databases are just the start. On-prem or cloud-native or hybrid? What and how to automate? What tools to pick for IaC?  What's the frontend framework of the month? Have you taken proper care of security? How about privacy? Did you think about accessibility? And finally, how is the project going? Communication working? Doing the right things in the right order?
```
[the remark this answers is not in the document]
Comment (crystoll) [reply _posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md:20]: Ah, Infrastructure as Code. Not relevant to open up here, just a quick list of technical stuff as an example of the width of things to master.

**Pull request merged**